Refactor Report Alert

tonyxrmdavidson opened the pull request:
This PR is the first step to refactoring the report alert script It contains the following updates

- Removes the configuration from main to it's own module
- Removes the cache functionality from main to it's own module
- Intorduces a flag to facilitate not using ssl verification
- Refactors the remaining functionality on the main script
- Fixes the issue of the url for each failed test taking us to all tests instead of the relevent test

Testing

It has been tested by using the same filters on both the UI and command to produce the same results by running the command 'python ./report_alert.py prow --no-verify --attr version_installed=4.12.0-0.nightly-2025-01-25-135326 --status FAILED -n automated-release'

tonyxrmdavidson committed:
Refactor Report Alert

This PR is the first step to refactoring the report alert script
It contains the following updates
* Removes the configuration from main to it's own module
* Removes the cache functionality from main to it's own module
* Intorduces a flag to facilitate not using ssl verification
* Refactors the remaining functionality on the main script
* Fixes the issue of the url for each failed test taking us to all tests instead of the relevent test.

It has been tested by using the  same filters on both the UI and command to produce the same results by running the command 'python ./report_alert.py prow --no-verify --attr version_installed=4.12.0-0.nightly-2025-01-25-135326 --status FAILED -n automated-release'

Signed-off-by: tonyxrmdavidson <tonyxrmdavidson@yahoo.co.uk>

tonyxrmdavidson committed:
Merge pull request #1 from tonyxrmdavidson/reportAlertRefactor

Refactor Report Alert

tonyxrmdavidson committed:
Refactor Report Alert

This PR is the first step to refactoring the report alert script It contains the following updates

- Removes the configuration from main to it's own module
- Removes the cache functionality from main to it's own module
- Intorduces a flag to facilitate not using ssl verification
- Refactors the remaining functionality on the main script
- Fixes the issue of the url for each failed test taking us to all tests instead of the relevent test

Testing

It has been tested by using the same filters on both the UI and command to produce the same results by running the command 'python ./report_alert.py prow --no-verify --attr version_installed=4.12.0-0.nightly-2025-01-25-135326 --status FAILED -n automated-release'

Signed-off-by: tonyxrmdavidson <tonyxrmdavidson@yahoo.co.uk>

tonyxrmdavidson committed:
This commit removes report portal client from the main script and creates a module for it

Testing

It has been tested using the same command as the initial commit and successfully completed returning the correct amount of launches, suites and tests

Signed-off-by: tonyxrmdavidson <tonyxrmdavidson@yahoo.co.uk>

yash2189 commented:
When I tried to run with the same command on Fedora I can still see verify_ssl being set as `True`
If I am not wrong it should set it `False`? 
```
(venv) [yajgaonk@fedora reportportal_alert]$ python ./report_alert.py prow --no-verify --attr version_installed=4.12.0-0.nightly-2025-01-25-135326 --status FAILED -n au
tomated-release 
INFO:root:Config loaded: base_url=https://reportportal-openshift.apps.ocp-c1.prod.psi.redhat.com, verify_ssl=True
```
However, If I change the config.json file by adding a key called "verify_ssl" and still pass the` no-verify ` it sets to False
I am unsure if it is working properly?
```
(venv) [yajgaonk@fedora reportportal_alert]$ python ./report_alert.py prow --no-verify --attr version_installed=4.12.0-0.nightly-2025-01-25-135326 --status FAILED -n automated-release 
INFO:root:Config loaded: base_url=https://reportportal-openshift.apps.ocp-c1.prod.psi.redhat.com, verify_ssl=False
INFO:root:Filters being applied for launches: {'filter.!cnt.name': 'automated-release', 'filter.eq.status': 'FAILED', 'filter.has.compositeAttribute': 'version_installed:4.12.0-0.nightly-2025-01-25-135326'}
```

tonyxrmdavidson committed:
Remove SSL_VERIFY key in config_module.py

This commit removes the key/value pair refering to SSL_Verification. It is no longer needed as this is now handled by a cli flag

Signed-off-by: tonyxrmdavidson <tonyxrmdavidson@yahoo.co.uk>

tonyxrmdavidson committed:
Merge branch 'main' into refactorReportAlert

yash2189 commented:
The above issue has been resolved related to `SSL_VERIFY_KEY`
In latest logs it has been removed
Eg:
`(venv) [yajgaonk@fedora reportportal_alert]$ python ./report_alert.py prow --no-verify --attr version_installed=4.12.0-0.nightly-2025-01-25-135326 --status FAILED -n automated-release --reset-cache
INFO:root:Cache reset successfully.
INFO:root:Config loaded: base_url=https://reportportal-openshift.apps.ocp-c1.prod.psi.redhat.com
INFO:root:Filters being applied for launches: {'filter.!cnt.name': 'automated-release', 'filter.eq.status': 'FAILED', 'filter.has.compositeAttribute': 'version_installed:4.12.0-0.nightly-2025-01-25-135326'}`

tonyxrmdavidson committed:
This adds functionality to filter by test name

I has been tested by filtering for nstaller within a known report portal filter and returned the correct of number of failed Installer tests

Signed-off-by: tonyxrmdavidson <tonyxrmdavidson@yahoo.co.uk>

tonyxrmdavidson commented:
@yash2189 @san7ket I have updated the scripts to use the --test-name flag for filtering on the client side once the json is returned.  Can you have a look at at it, run it so we can get this merged and continue with the other work for it.

tonyxrmdavidson committed:
This commit pushes the change to filter with part name

Signed-off-by: tonyxrmdavidson <tonyxrmdavidson@yahoo.co.uk>

yash2189 commented:
LGTM

san7ket commented:
Looks good
